streams/src/main/java/org/apache/kafka/streams/TopologyConfig.java

```diff
@@ -151,6 +154,7 @@ public TopologyConfig(final String topologyName, final StreamsConfig globalAppCo
 
         this.applicationConfigs = globalAppConfigs;
         this.topologyOverrides = topologyOverrides;
+        this.processingExceptionHandlerSupplier = () -> globalAppConfigs.getConfiguredInstance(PROCESSING_EXCEPTION_HANDLER_CLASS_CONFIG, ProcessingExceptionHandler.class);
```

Comment:
Why do we need to use a supplier here?

Reply:
Because Streams creates a processing exception handler per task.
See https://github.com/apache/kafka/blob/a5bfc2190c3448039c9361909e547f64f7fdb6e2/streams/src/main/java/org/apache/kafka/streams/TopologyConfig.java#L288

Reply:
Sure, but line 288 could just be
```
globalAppConfigs.getConfiguredInstance(PROCESSING_EXCEPTION_HANDLER_CLASS_CONFIG, ProcessingExceptionHandler.class
```
Not sure why we _need_ the supplier indirection?

For the existing `timestampExtractorSupplier` and `deserializationExceptionHandlerSupplier` is a little different because we setup different suppliers for different cases (so using suppliers simplifies the code), but for the new one we have only one supplier and thus it seem unnecessary.

No need to change anything. We follow an established pattern, but it just was confusing to me, as it seems (strictly speaking) unnecessary to have the layer of indirection.

Reply:
`processingExceptionHandler` has been created following the `deserializationExceptionHandler`. Thus a supplier has been used. 

The supplier can be removed to make the code clearer if needed.

Reply:
Yes. Understood. -- I think it's not necessary to have a supplier, but also ok to keep the code as-is.